### PR TITLE
move from 0.3.1 to 0.3.3 in the CI test

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

the v0.3.1 is confusing ,we should update to latest version

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -316,16 +316,16 @@ create-cluster: $(KUSTOMIZE) $(ENVSUBST) ## Create a development Kubernetes clus
 	mkdir ~/.cluster-api
 	chmod +x $(CLUSTERCTL)
 	# Create clusterctl.yaml to use local OpenStack provider
-	mkdir -p ./out/infrastructure-openstack/v0.3.1
+	mkdir -p ./out/infrastructure-openstack/pr
```
Comment (sbueringer) [inline Makefile:319]: what does pr stand for? I think the reason I added it might have been because clusterctl expected it in this folder. But not sure

Comment (jichenjc) [reply Makefile:319]: yes, my test showed that ,but I think v0.3.1 does not make sense now... at least need move to 0.3.2 or 0.3.3
checking the reason and dependency you mentioned above now..

Comment (sbueringer) [reply Makefile:319]: Yup. Let's see what we need here :). I don't really know anymore where the dependency came from if there is one

Comment (hidekazuna) [reply Makefile:319]: FYI: CAPA uses next minor version: v0.7.0
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/clusterctl-settings.json#L5

Comment (jichenjc) [reply Makefile:319]: Error: failed to get provider components for the "openstack" provider: failed to get repository client for the InfrastructureProvider with name openstack: error creating the local filesystem repository client: invalid version: "pr". Version must obey the syntax and semantics of the "Semantic Versioning" specification (http://semver.org/) and path format {basepath}/{provider-name}/{version}/{components.yaml}

this is the issue you mentioned @sbueringer 
so we need switch to 0.3.3 and update it whenever we have a new version ...

**Comment by jichenjc** [comment]
recheck

**Comment by jichenjc** [comment]
recheck

**Comment by jichenjc** [comment]
recheck

**Comment by jichenjc** [comment]
@hidekazuna can you help to check this? Thanks

**Comment by hidekazuna** [comment]
Every time we release minor version, we will update this file. How about using 0.4.0 to avoid update so often?

**Comment by jichenjc** [comment]
> Every time we release patch version, we will update this file. How about using 0.4.0 to avoid update so often?

clusterctl need this version as previous comments mentioned, clusterctl will create from desired version (0.3.1 , 0.3.3 etc)
0.4.0 is not ready yet..

**Comment by hidekazuna** [comment]
> 
> 
> > Every time we release patch version, we will update this file. How about using 0.4.0 to avoid update so often?
> 
> clusterctl need this version as previous comments mentioned, clusterctl will create from desired version (0.3.1 , 0.3.3 etc)
> 0.4.0 is not ready yet..

0.4.0 is semver. master branch is for future version, so that it is not odd to be 0.4.0. Actually CAPA uses future version.

**Comment by jichenjc** [comment]
no ,  0.4.0 will not work , because clusterctl will load <version> from docker registry and 0.4.0 not exist now

**Comment by hidekazuna** [comment]
Still I do not understand. CAPO image in CI is tagged with another tag, which is not v0.3.3.
We uses local CAPO for create-cluster target in Makefile.

In my local environment,

```bash
echo "  minor: 3" >> ./out/infrastructure-openstack/$(CAPO_VERSION)/metadata.yaml
```

must be

```bash
echo "  minor: 4" >> ./out/infrastructure-openstack/$(CAPO_VERSION)/metadata.yaml
```

**Comment by jichenjc** [comment]
Error: failed to get provider components for the "openstack" provider: failed to get repository client for the InfrastructureProvider with name openstack: error creating the local filesystem repository client: invalid version: "pr". Version must obey the syntax and semantics of the "Semantic Versioning" specification (http://semver.org/) and path format {basepath}/{provider-name}/{version}/{components.yaml}

and please check our CI 

https://logs.openlabtesting.org/logs/86/686/6cbe2fd1b07402d4593376b55f9d48fdef6fdaec/check/cluster-api-provider-openstack-current-acceptance-test-v1.18/dcf3097/job-output.txt.gz

2020-12-10 08:47:40.622041 | ubuntu-bionic-large | Installing Provider="infrastructure-openstack" Version="0.3.3" TargetNamespace="capo-system"


the 0.3.3 is comes from {version} above ..

**Comment by hidekazuna** [comment]
> Error: failed to get provider components for the "openstack" provider: failed to get repository client for the InfrastructureProvider with name openstack: error creating the local filesystem repository client: invalid version: "pr". Version must obey the syntax and semantics of the "Semantic Versioning" specification (http://semver.org/) and path format {basepath}/{provider-name}/{version}/{components.yaml}

"pr" is not semver. 0.4.0 is semver. Check out  ["Semantic Versioning" specification](http://semver.org/)

> and please check our CI
> 
> https://logs.openlabtesting.org/logs/86/686/6cbe2fd1b07402d4593376b55f9d48fdef6fdaec/check/cluster-api-provider-openstack-current-acceptance-test-v1.18/dcf3097/job-output.txt.gz
> 
> 2020-12-10 08:47:40.622041 | ubuntu-bionic-large | Installing Provider="infrastructure-openstack" Version="0.3.3" TargetNamespace="capo-system"
> 
> the 0.3.3 is comes from {version} above ..

This is because you created 0.3.3 directory. 
2020-12-10 08:47:01.732747 | ubuntu-bionic-large | hack/tools/bin/kustomize build config > ./out/infrastructure-openstack/0.3.3/infrastructure-components.yaml

If you create 0.4.0 directory, you will get the following like my local environment.

```bash
ubuntu@capi:~/cluster-api-provider-openstack$ clusterctl init --infrastructure openstack --core cluster-api:v${CAPI_VERSION} --bootstrap kubeadm:v${CAPI_VERSION} --control-plane kubeadm:v${CAPI_VERSION}
Fetching providers
Installing cert-manager Version="v0.16.1"
Waiting for cert-manager to be available...
Installing Provider="cluster-api" Version="v0.3.11" TargetNamespace="capi-system"
Installing Provider="bootstrap-kubeadm" Version="v0.3.11" TargetNamespace="capi-kubeadm-bootstrap-system"
Installing Provider="control-plane-kubeadm" Version="v0.3.11" TargetNamespace="capi-kubeadm-control-plane-system"
Installing Provider="infrastructure-openstack" Version="0.4.0" TargetNamespace="capo-system"
```

**Comment by hidekazuna** [comment]
Trying to save time 😃   #712

**Comment by jichenjc** [comment]
>>>Installing Provider="infrastructure-openstack" Version="0.4.0" TargetNamespace="capo-system"

do we have 0.4 capo in hand?

**Comment by hidekazuna** [comment]
> > > > Installing Provider="infrastructure-openstack" Version="0.4.0" TargetNamespace="capo-system"
> 
> do we have 0.4 capo in hand?

we load capo docker image into kind cluster:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/Makefile#L333-L336

**Comment by jichenjc** [comment]
> > > > > Installing Provider="infrastructure-openstack" Version="0.4.0" TargetNamespace="capo-system"
> > 
> > 
> > do we have 0.4 capo in hand?
> 
> we load capo docker image into kind cluster:
> https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/Makefile#L333-L336

understand, so we should keep current 0.3.1 ? at least it looks weird and confusing when debug..

**Comment by hidekazuna** [comment]
0.3.1 is already released. CI is for master, master is to develop 0.4.0. So I like 0.4.0.

**Comment by jichenjc** [comment]
@hidekazuna updated just now ,thanks~

File: Makefile [hunk 2]
```diff
@@ -316,16 +317,16 @@ create-cluster: $(KUSTOMIZE) $(ENVSUBST) ## Create a development Kubernetes clus
...
+	echo "  url: $(PWD)/out/infrastructure-openstack/$(CAPO_VERSION)/infrastructure-components.yaml" >> ./out/clusterctl.yaml
 	echo "  type: InfrastructureProvider" >> ./out/clusterctl.yaml
 
-	echo "releaseSeries:" > ./out/infrastructure-openstack/v0.3.1/metadata.yaml
-	echo "- major: 0" >> ./out/infrastructure-openstack/v0.3.1/metadata.yaml
-	echo "  minor: 3" >> ./out/infrastructure-openstack/v0.3.1/metadata.yaml
-	echo "  contract: v1alpha3" >> ./out/infrastructure-openstack/v0.3.1/metadata.yaml
+	echo "releaseSeries:" > ./out/infrastructure-openstack/$(CAPO_VERSION)/metadata.yaml
+	echo "- major: 0" >> ./out/infrastructure-openstack/$(CAPO_VERSION)/metadata.yaml
+	echo "  minor: 3" >> ./out/infrastructure-openstack/$(CAPO_VERSION)/metadata.yaml
```
Comment (hidekazuna) [inline Makefile:328]: we need to update to 4.

**Comment by sbueringer** [comment]
/lgtm

just fyi. cluster-api released 0.3.12 today: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v0.3.12

**Comment by sbueringer** [comment]
/hold

feel free to /hold cancel when you want to merge

**Comment by jichenjc** [comment]
cluster api 0.3.12 is something we might catch up later on  (will take a look later on)
so far we can focus on this issue , thanks both of you comments

 /hold cancel

**Pull request merged**